components/brave_news/browser/topics_fetcher.h

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#ifndef BRAVE_COMPONENTS_BRAVE_NEWS_BROWSER_TOPICS_FETCHER_H_
+#define BRAVE_COMPONENTS_BRAVE_NEWS_BROWSER_TOPICS_FETCHER_H_
+
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "base/containers/flat_map.h"
+#include "base/functional/callback_forward.h"
+#include "brave/components/api_request_helper/api_request_helper.h"
+#include "brave/components/brave_news/api/topics.h"
+#include "services/network/public/cpp/shared_url_loader_factory.h"
+
+namespace brave_news {
+
+constexpr char kTopicsEndpoint[] = "/news-topic-clustering/topics";
+constexpr char kTopicArticlesEndpoint[] = "/news-topic-clustering/topics_news";
+
+using Topic =
```

Comment:
nit: I feel this alias being `Topic` is a bit confusing, as in fact it is a pair of topic/articles. I don't feel strongly about this, but just flagging if you feel like there could be some better name.

Reply:
I agree, it's a bit confusing - I did consider `TopicArticles` but I think that's a bit worse :laughing: I could rename the IDL `Topic` to be `TopicDetails` and that might make things a bit more obvious? A `Topic` would be a `TopicDetails` + `std::vector<TopicArticles>`?

Reply:
I think TopicAndArticles could make sense when we think of something like

```
Topics topics;
TopicAndArticles topic_and_articles = topics.front();
const [topic, articles] = topic_and_articles;
``

Reply:
Yeah, I think the name `Topic` in the IDL, as `Detail` actually doesn't add anything concrete to it.

Reply:
`TopicAndArticles` it is :laughing: